tempesta_fw/cache.c

```diff
@@ -950,8 +972,52 @@ cache_req_process_node(TfwHttpReq *req, unsigned long key,
 out:
 	action(req, resp);
 
-	if (ce)
-		tdb_rec_put(ce);
+	tfw_cache_dbce_put(ce);
+}
+
+/*
+ * Invalidate a cache entry.
+ */
+static int
+tfw_cache_purge_set_expired(TfwHttpReq *req, unsigned long key)
+{
+	TdbIter iter;
+	TDB *db = node_db();
+	TfwCacheEntry *ce = NULL;
+
+	if (!(ce = tfw_cache_dbce_get(db, &iter, req, key))) {
+		tfw_cache_dbce_put(ce);
+		return -ENOENT;
+	}
+
+	/* ce->lifetime = 0; */
+
+	tfw_cache_dbce_put(ce);
+	return 0;
+}
+
+/*
+ * Process PURGE request method according to the configuration.
+ */
+static int
+tfw_cache_purge_method(TfwHttpReq *req, unsigned long key)
+{
+	TfwVhost *vhost = tfw_vhost_get_default();
+
+	if (!vhost->cache_purge)
+		return tfw_http_send_403((TfwHttpMsg *)req);
+
+	if (vhost->cache_purge_acl) {
```

Comment:
Let's deny PURGE request by default: if admin forgot to set configure it he figure it out quickly otherwise default accepting PURGE request can open security flaws.